README.md

```diff
@@ -1,7 +1,34 @@
 ## What is the difference between push, pull, and fetch?
 
-- `git push` - sent changes from a local branch to a remote repo
-- `git fetch` - get changes from a remote repo into your tracking branch
-- `git pull` - will get changes from a remote branch into your tracking branch and merge them into a local branch
+- `git fetch` - downloads changes from your _remote_ branch, and adds them to your
+  _tracking_ branch
+- `git pull` - downloads changes from your _remote_ branch, adds them to your
+  _tracking_ branch, and then merges them into your _tracking_ branch
+- `git push` - uploads changes from your _local_ or _tracking_ branch to your
+  _remote_ branch
+
+Often `git push` and `git pull` are described as equivalent. This isn't entirely
+correct, since `git pull` does two things.
+
+The `git fetch` command takes your current branch, and checks to see if there is
+a _tracking_ branch. If so, it looks for changes in the _remote_ branch. If there
+are changes, they are downloaded, but not merged into your _tracking_ branch. You
+must manually issue the `git merge origin/main` command (for the "main" branch)
```

Comment:
I changed references from master to main which is our new guideline, but also this section is confusing to me because of the comment above ☝️ .  I think what you are trying to get at is "Here's how to merge the current state of main into your feature branch.", but there are some steps missing if that's the case. Also, what if they are tracking on a shared feature branch?

Finally, this last sentence with the hyphen has some structural issues and should probably be reworded. Maybe we should have a quick zoom so I can better understand what you are going for.